src/lib/circuit.ts

```diff
@@ -0,0 +1,221 @@
+    );
+  }
+
+  // utility namespace
+
+  /**
+   * Create a new witness. A witness, or variable, is a value that is provided as input
+   * by the prover. This provides a flexible way to introduce values from outside into the circuit.
+   * However, note that nothing about how the value was created is part of the proof - `Circuit.witness`
+   * behaves exactly like user input. So, make sure that after receiving the witness you make any assertions
+   * that you want to associate with it.
+   *
+   * Example for re-implementing `Field.inv` with the help of `witness`:
+   * ```
+   * let invX = Circuit.witness(Field, () => {
+   *   // compute the inverse of `x` outside the circuit, however you like!
+   *   return Field.inv(x));
+   * }
+   * // prove that `invX` is really the inverse of `x`:
+   * invX.mul(x).assertEquals(1);
+   * ```
+   */
+  static witness = SnarkyCircuit.witness;
+  /**
+   * Runs code as a prover.
+   */
+  static asProver = SnarkyCircuit.asProver;
+  /**
+   * Runs code and checks its correctness.
+   */
+  static runAndCheck = SnarkyCircuit.runAndCheck;
+  /**
+   * Returns information about the constraint system in the callback function.
+   */
+  static constraintSystem = SnarkyCircuit.constraintSystem;
+  /**
+   * Returns a low-level JSON representation of the `Circuit` from its {@link Keypair}:
+   * a list of gates, each of which represents a row in a table, with certain coefficients and wires to other (row, column) pairs
+   */
+  static constraintSystemFromKeypair =
+    SnarkyCircuit.constraintSystemFromKeypair;
+  /**
+   * Creates a {@link Provable} for a generic array.
+   */
+  static array = SnarkyCircuit.array;
+  /**
+   * Asserts that two values are equal.
+   */
+  static assertEqual = SnarkyCircuit.assertEqual;
+  /**
+   * Checks if two elements are equal.
+   */
+  static equal = SnarkyCircuit.equal;
+  /**
+   * Circuit-compatible if-statement.
+   */
+  static if = SnarkyCircuit.if;
+  /**
+   * Generalization of `Circuit.if` for choosing between more than two different cases.
+   * It takes a "mask", which is an array of `Bool`s that contains only one `true` element, as well as a type/constructor and an array of values of that type.
+   * The result is that value which corresponds to the true element of the mask. Example:
+   *
+   * ```ts
+   * let x = Circuit.switch([Bool(false), Bool(true)], Field, [Field(1), Field(2)]);
+   * x.assertEquals(2);
+   * ```
+   */
+  static switch = SnarkyCircuit.switch;
+  /**
+   * Serializes an element into {@link Field} elements.
+   */
+  static toFields = SnarkyCircuit.toFields;
+  /**
+   * Checks if the circuit is in prover mode.
+   */
+  static inProver = SnarkyCircuit.inProver;
+  /**
+   * Checks if the circuit is in checked computation mode.
+   */
+  static inCheckedComputation = SnarkyCircuit.inCheckedComputation;
+  /**
+   * Interface to log elements within a circuit. Similar to `console.log()`.
+   */
```

Comment:
```suggestion
  /**
   * Interface to log elements within a circuit. Similar to `console.log()`.
   * @example
   * ```ts
   * const element = Field(42);
   * Circuit.log(element);
   * ```
   */
```